lib/Controller/MailboxesController.php

```diff
@@ -248,8 +248,12 @@ public function update() {
 	#[TrapError]
 	public function create(int $accountId, string $name, array $specialUseAttributes = []): JSONResponse {
 		$account = $this->accountService->find($this->currentUserId, $accountId);
-
-		return new JSONResponse($this->mailManager->createMailbox($account, $name, $specialUseAttributes));
+		try {
+			$mailbox = $this->mailManager->createMailbox($account, $name, $specialUseAttributes);
+		} catch (ServiceException $e) {
+			return new JSONResponse([], Http::STATUS_BAD_REQUEST);
```

Comment:
This will now catch *any* service exception and return a nondescriptive  HTTP response. This would be painful to debug.

Do a simple in_array check comparing the sent special use to the allowed ones. Then carry on with the string and add type hints for the allowed values for the deeper service methods.